src/core/scope.ml

```diff
@@ -433,13 +442,16 @@ type pre_rule =
   ; pr_lhs     : term list
   (** Arguments of the LHS. *)
   ; pr_vars    : term_env Bindlib.mvar
-  (** Pattern variables that can appear in the RHS. *)
+  (** Pattern variables that can appear in the RHS. The [pr_xvars] last
+      variables do not appear in the LHS. *)
   ; pr_rhs     : tbox
   (** Body of the RHS, should only be unboxed once. *)
   ; pr_names   : (int, string) Hashtbl.t
   (** Gives the original name (if any) of pattern variable at given index. *)
   ; pr_arities : int array
-  (** Gives the arity of all the pattern varialbes in field [pr_vars]. *) }
+  (** Gives the arity of all the pattern varialbes in field [pr_vars]. *)
+  ; pr_xvars: int
+  (** Number of extra variable in [pr_vars]. *) }
```

Comment:
variable+s. Comment not clear: number of variables of the RHS not occurring in the LHS.